library/ssl_tls12_server.c

```diff
@@ -3204,6 +3212,14 @@ static int ssl_prepare_server_key_exchange( mbedtls_ssl_context *ssl,
 #endif
         int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
 
+        if( dig_signed == NULL )
+        {
+            MBEDTLS_SSL_DEBUG_MSG( 1, ( "should never happen" ) );
+            return( MBEDTLS_ERR_SSL_INTERNAL_ERROR );
+        }
```

Comment:
Please move this check before the declaration of `dig_signed_len`, and initialize it directly to its intended value. We're writing C99 now, and can take advantage of mid-block declarations when no end-of-function cleanup needs to be done.